### PR TITLE
drivers: sensor: Add initial support for hm330x particle sensor

### PR DESCRIPTION
**Pull request by kartben**
Adds initial support for hm330x dust sensor series. 
Allows to read PM1, PM2.5 and PM10 concentrations in "atmospheric environment".
A further update to the driver may add support for also reading "standard" CF1 concentrations by exposing of a custom sensor attribute or a Kconfig option.

Tested with Grove - Laser PM2.5 Sensor (HM3301) attached to a Wio Terminal.

### CODE REVIEW

**Review by MaureenHelm — CHANGES_REQUESTED** [review]
Some small nits, but otherwise LGTM

File: dts/bindings/sensor/seeed,hm330x.yaml [hunk 1]
```diff
@@ -0,0 +1,9 @@
+# Copyright (c) 2023 benjamin@zephyrproject.org
+# SPDX-License-Identifier: Apache-2.0
+
+description: |
+    Seeed Studio HM330x dust particle sensor
+
+compatible: "seeed,hm330x"
+
+include: [i2c-device.yaml]
```
[the remark this answers is not in the document]
Comment (kartben) [reply dts/bindings/sensor/seeed,hm330x.yaml]: oops.

File: drivers/sensor/hm330x/hm330x.h [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+
+struct hm330x_data {
+	uint16_t pm_1_0_sample;
+	uint16_t pm_2_5_sample;
+	uint16_t pm_10_sample;
+};
+
+struct hm330x_config {
+	struct i2c_dt_spec i2c;
+};
```
[the remark this answers is not in the document]
Comment (kartben) [reply drivers/sensor/hm330x/hm330x.h]: +1, makes sense. and done.

**Comment by kartben** [comment]
> Some small nits, but otherwise LGTM

Thanks for the quick review! I've addressed all your comments.

**Review by MaureenHelm — APPROVED** [review]

**Review by carlescufi — APPROVED** [review]
LGTM!

**Pull request merged**